quiz/quiz.go

```diff
@@ -0,0 +1,168 @@
+/*
+	quiz defines a quiz and its questions and provides methods for starting and processing quizzes as well as reading in
+    a Quiz from file.
+*/
+package quiz
+
+import (
+	"bufio"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"strings"
+	"time"
+)
+
+// A Quiz struct is a collection of up to 100 Questions, the name of the file containing the quiz, the number of
+// questions in the quiz and the number of questions the user got correct.
+type Quiz struct {
+	Questions        []Question
+	Filename         string
+	TotalQuestions   int
+	CorrectQuestions int
+}
+
+// NewQuiz creates a new Quiz struct based on the information in the file at the given file name
+func NewQuiz(filename string) (*Quiz, error) {
+	fmt.Println("\nGetting quiz using file:\t", filename)
+
+	// open the file and save Question to a Quiz struct
+	csvFile, err := os.Open(filename)
+
+	if err != nil {
+		fmt.Println("Error opening CSV file:\t", err)
+		return nil, err
+	}
+
+	defer func() {
+		if csvErr := csvFile.Close(); csvErr != nil {
+			fmt.Println(csvErr)
+		}
+	}()
+
+	reader := csv.NewReader(bufio.NewReader(csvFile))
```

Comment:
This might be getting away from the original intent of the exercise but there is a go package that will unmarshall csv files into structs: https://github.com/gocarina/gocsv.

Reply:
That looks really useful, but I think you're right that it is outside the original intent. The README specifically calls out the `csv` package.